### PR TITLE
Fix launchers

### PR DESCRIPTION
**Pull request by AlexReimann**
Add test for launchers and fix them

Side note:
Seems like the rostests get executed in order, except the last test gets execute before all others.. I wonder why / how

### CODE REVIEW

**Comment by asmodehn** [comment]
How about changing the existing tests to use the launchers, instead of adding specific tests ?

**Comment by AlexReimann** [comment]
Do you mean:
1. Calling the launchers in the python script 
2. Use the launchers in the ``.test`` file and remove the node launching from the python script

In case of 2. the test scripts can't be run alone and need to always be started with the launchers.
No idea how to do 1. and if that is a good idea.

**Comment by asmodehn** [comment]
I was thinking of 2, without removing the node launching from python script...
But that makes more problems (arguments that becomes yaml files...)

So I thought maybe it s better to have specific integration tests, testing launch files and yaml params...

**Comment by AlexReimann** [comment]
I merged the master into here to make the tests succeed.

Now I get following errors when launching ``roslaunch ros1_pytemplate all.launch --screen``:

```
[ERROR] [WallTime: 1502331806.914518] can't locate node [echo.sh] in package [ros1_pytemplate]
[ERROR] [WallTime: 1502331806.914604] ERROR: cannot launch node of type
[ros1_pytemplate/echo.sh]: can't locate node [echo.sh] in package [ros1_pytemplate]
ERROR: cannot launch node of type [ros1_pytemplate/echo.sh]: can't locate node [echo.sh] in
package [ros1_pytemplate]
```

**Comment by AlexReimann** [comment]
Fixed some stuff and added dependency to msgs package.

Still getting python errors in the ``test_fibonacci.py``:

```
[ros1_pytemplate:make] Traceback (most recent call last):                                                                                                                                                                                    
[ros1_pytemplate:make]   File "/opt/groot/others_ws/src/ros1_template/ros1_pytemplate/rostests/integration/test_fibonacci.py", line 18, in <module>                                                                                          
[ros1_pytemplate:make]     pyros_setup.configurable_import().configure().activate()                                                                                                                                                          
[ros1_pytemplate:make] AttributeError: 'module' object has no attribute 'configurable_import'                                                                                                                                                
[ros1_pytemplate:make] Traceback (most recent call last):                                                                                                                                                                                    
[ros1_pytemplate:make]   File "/opt/groot/others_ws/src/ros1_template/ros1_pytemplate/nodes/fibonacci_sub_node.py", line 64, in <module>                                                                                                     
[ros1_pytemplate:make]     rospy.init_node('fibonacci_sub_node')                                                                                                                                                                             
[ros1_pytemplate:make]   File "/opt/ros/indigo/lib/python2.7/dist-packages/rospy/client.py", line 320, in init_node                                                                                                                          
[ros1_pytemplate:make]     raise rospy.exceptions.ROSInitException("init_node interrupted before it could complete")                                                                                                                         
[ros1_pytemplate:make] rospy.exceptions.ROSInitException: init_node interrupted before it could complete                                                                                                                                     
[ros1_pytemplate:make] Traceback (most recent call last):                                                                                                                                                                                    
[ros1_pytemplate:make]   File "/opt/groot/others_ws/src/ros1_template/ros1_pytemplate/nodes/fibonacci_pub_node.py", line 60, in <module>                                                                                                     
[ros1_pytemplate:make]     rospy.init_node('fibonacci_pub_node', )                                                                                                                                                                           
[ros1_pytemplate:make]   File "/opt/ros/indigo/lib/python2.7/dist-packages/rospy/client.py", line 320, in init_node                                                                                                                          
[ros1_pytemplate:make]     raise rospy.exceptions.ROSInitException("init_node interrupted before it could complete")                                                                                                                         
[ros1_pytemplate:make] rospy.exceptions.ROSInitException: init_node interrupted before it could complete                                                                                                                                     
[ros1_pytemplate:make] testtest_fibonacci ... FAILURE!                                                                                                                                                                                       
[ros1_pytemplate:make] FAILURE: test [test_fibonacci] did not generate test results                                                                                                                                                          
[ros1_pytemplate:make]   File "/usr/lib/python2.7/unittest/case.py", line 331, in run                                                                                                                                                        
[ros1_pytemplate:make]     testMethod()                                                                                                                                                                                                      
[ros1_pytemplate:make]   File "/opt/ros/indigo/lib/python2.7/dist-packages/rostest/runner.py", line 164, in fn                                                                                                                               
[ros1_pytemplate:make]     self.assert_(os.path.isfile(test_file), "test [%s] did not generate test results"%test_name)                                                                                                                      
[ros1_pytemplate:make]   File "/usr/lib/python2.7/unittest/case.py", line 424, in assertTrue                                                                                                                                                 
[ros1_pytemplate:make]     raise self.failureException(msg) 
```

Can you take a look at that? Maybe just some import / dependency missing?

**Comment by asmodehn** [comment]
Ah pyros_setup was me playing with my editor to launch rostests... I m still trying to find a clean way that doesnt break other people computer, so I think we can remove that for now.

But it looks like you are using an old version (the indigo package has been EOL btw. it s only needed when running from python virtualenv)

**Comment by asmodehn** [comment]
The build failing seems to be timing issue... any awesome idea to not have hardcoded timeout and always work no matter how long it can take  (but not forever) ?

**Comment by AlexReimann** [comment]
I did some testing and it seems the message never arrives. I guess it is not latched and the subscriber starts after it is published?

The 5 secs should be more than enough

**Comment by asmodehn** [comment]
I think it s because of timeout, because it works on my local setup : 
```
(pip_template)alexv@alexv-indigo-vbox:/opt/groot/bootstrap_ws/src/ros1_template/ros1_pytemplate$ rostest ros1_pytemplate fibonacci_sub.test 
... logging to /home/alexv/.ros/log/2017-08-09/rostest-alexv-indigo-vbox-29311.log
[ROSUNIT] Outputting test results to /home/alexv/.ros/test_results/ros1_pytemplate/rostest-rostests_fibonacci_pubsub_fibonacci_sub.xml
testtest_fibonacci_sub ... ok

[ROSTEST]-----------------------------------------------------------------------

[ros1_pytemplate.rosunit-test_fibonacci_sub/test_follower][passed]

SUMMARY
 * RESULT: SUCCESS
 * TESTS: 1
 * ERRORS: 0
 * FAILURES: 0

rostest log file is in /home/alexv/.ros/log/2017-08-09/rostest-alexv-indigo-vbox-29311.log

```

**Comment by AlexReimann** [comment]
It also works most of the time for me, but I would say 1 out of 10 it doesn't work. Which was why I thought it is a timing issue.

**Comment by AlexReimann** [comment]
[There is](http://wiki.ros.org/rostest/Nodes#Subscribed_Topics) a `wait_time` parameter for the tests.

Should test that maybe? But it says default is 20.0 sec .. so no idea

**Comment by asmodehn** [comment]
Found 'a' problem, probably related. rostest is launching the "main" section (at least on indigo), but I thought it wouldn't... 

I was expecting rostest to "use" the unittest.TestCase in the test module, but it looks like rostest creates his own test case : https://github.com/ros/ros_comm/blob/lunar-devel/tools/rostest/src/rostest/rostest_main.py#L159 which mess up stuff...

So that means that nodes would be created twice with our current code... I ll fix it (I think I remember doing that two years ago already...)

**Comment by asmodehn** [comment]
There are the hacks I used to have in tests for python https://github.com/pyros-dev/pyros-utils/blob/devel/pyros_utils/rostest_nose.py 
I think I ll use them also in the basic pytemplate as well

**Comment by asmodehn** [comment]
I kept test here, as per ROS tutorials. also tested launching nodes in different namespaces to illustrate remap.

**Review by asmodehn — APPROVED** [review]
LGTM

File: ros1_pytemplate/rostests/answer_server/test_answer_server.py [hunk 1]
```diff
@@ -9,11 +9,11 @@
 # Imports
 ##############################################################################
 
+import sys
+import unittest
+
 import rospy
 import rostest
-import unittest
```
Comment (AlexReimann) [inline ros1_pytemplate/rostests/answer_server/test_answer_server.py:14]: Are you using some tool to adjust the imports automatically?

File: ros1_pytemplate/launch/fibonacci.launch [hunk 2]
```diff
@@ -4,17 +4,18 @@
 <launch>
 
   <!-- Args -->
-  <arg name="config_override"        default="$(find ros1_template)/config/fibonacci-override.yaml" doc="used by upper launchers to customize configuration"/>
+  <arg name="config_override"        default="$(find ros1_pytemplate)/config/fibonacci-override.yaml" doc="used by upper launchers to customize configuration"/>
 
-  <!-- Launch oracle node -->
+  <!-- Launch prophet node -->
```
Comment (AlexReimann) [inline ros1_pytemplate/launch/fibonacci.launch:9]: Maybe just call it publisher?

File: ros1_pytemplate/rostests/answer_server/test_answer_server.py (hunk 1 above)
Comment (asmodehn) [reply ros1_pytemplate/rostests/answer_server/test_answer_server.py:14]: My IDE shows me when the imports are unused and therefore not needed

File: ros1_pytemplate/CMakeLists.txt [hunk 3]
```diff
@@ -22,10 +22,12 @@ find_package(catkin REQUIRED
 
 catkin_python_setup()
 
-# Lint Python modules
-file(GLOB_RECURSE ${PROJECT_NAME}_PY_SRCS
+if (roslint_FOUND)
```
Comment (AlexReimann) [inline ros1_pytemplate/CMakeLists.txt:25]: Who sets this?

Comment (asmodehn) [reply ros1_pytemplate/CMakeLists.txt:25]: This is basic cmake  find_package()

Comment (AlexReimann) [reply ros1_pytemplate/CMakeLists.txt:25]: I see, me n00b, me sorry :>

File: ros1_pytemplate/launch/fibonacci.launch (hunk 2 above)
Comment (AlexReimann) [reply ros1_pytemplate/launch/fibonacci.launch:9]: Besides this can merge

**Pull request merged**